### PR TITLE
Docs: Add instructions for 'esptool' by Christian Klippel.

### PR DESCRIPTION
**Pull request by Arnie97**
- [x] This PR is compliant with the [contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md)
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/en/*`.

Committers supporting this PR: leave blank
Add instructions for 'esptool' by Christian Klippel.

### CODE REVIEW

File: docs/en/flash.md [hunk 1]
```diff
@@ -15,13 +15,33 @@ Supported platforms: OS X, Linux, Windows, anything that runs Python
...
 Check the [esptool flash modes documentation](https://github.com/themadinventor/esptool#flash-modes) for details and other options.
 
+## esptool-ck
+> Esptool reads the compiled program in ELF format, extracts code and data sections, and either dumps a section to a file or assembles the firmware file from several segments. Esptool also communicates with the bootloader to upload firmware files to flash. Esptool can automatically put the board into UART bootloader mode using a variety of methods.
+
+Source: [https://github.com/igrr/esptool-ck](https://github.com/igrr/esptool-ck)
+
+Supported platforms: Windows, OS X, Linux, or any other POSIX compatible OS with a C compiler. If you're a Windows or Linux user, you can also [download the binary release here](https://github.com/igrr/esptool-ck/releases).
+
+Run the following command to flash an combined binary image.
```
Comment (devsaurus) [inline docs/en/flash.md]: Please use the same wording *aggregated* like above to avoid confusion about differences.

File: docs/en/flash.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
-Adafruit provides a really nice [firmware flashing tutorial](https://learn.adafruit.com/building-and-running-micropython-on-the-esp8266/flash-firmware). Below you'll find just the basics for the two popular tools esptool and NodeMCU Flasher.
+Adafruit provides a really nice [firmware flashing tutorial](https://learn.adafruit.com/building-and-running-micropython-on-the-esp8266/flash-firmware). Below you'll find just the basics for three popular tools: esptool.py, esptool-ck and NodeMCU Flasher.
 
 !!! attention
 
     Keep in mind that the ESP8266 needs to be [put into flash mode](#putting-device-into-flash-mode) before you can flash a new firmware!
 
-## esptool
+## esptool.py
```
Comment (marcelstoer) [inline docs/en/flash.md]: Riiiight...while the GitHub repo is called `esptool` the author(s) mostly seem to be referring to it as `esptool.py`.

File: docs/en/flash.md [hunk 3]
```diff
@@ -15,13 +15,33 @@ Supported platforms: OS X, Linux, Windows, anything that runs Python
 
 Run the following command to flash an *aggregated* binary as is produced for example by the [cloud build service](build.md#cloud-build-service) or the [Docker image](build.md#docker-image).
 
-`esptool.py --port <USB-port-with-ESP8266> write_flash -fm <mode> -fs <size> 0x00000 <nodemcu-firmware>.bin`
+`esptool.py --port <serial-port-of-ESP8266> write_flash -fm <mode> -fs <size> 0x00000 <nodemcu-firmware>.bin`
```
Comment (marcelstoer) [inline docs/en/flash.md]: Slightly embarrassing that silly glitch has been in our docs for so long.

**Comment by marcelstoer** [comment]
Thanks for the nice contribution but I'm not 100% I want to have this tool mentioned in our docs. 
- I have some doubt about the popularity of esptool-ck. Besides esptool.py and NodeMCU Flasher there a _a number_ of other flashing tools. esptool.py and NodeMCU Flasher are AFAICT by far the most popular ones. If we document a non-top-2 tool we'd probably have to document others as well (and keep up with them). This could partically be fixed by rephrasing the intro paragraph if we wanted to have esptool-ck mentioned.
- We might have more support effort due to people mixing up esptool-ck and esptool.py.

**Comment by marcelstoer** [comment]
I incorporated the minor fixes but since no other committer came forward to agree/disagree with me as for the inclusion of esptool-ck I'm going to close this. Fellow committers, feel free to reopen if you disagree.

**Pull request closed without merging**